### PR TITLE
Added edit distance global constraint propagator

### PR DESCRIPTION
**Pull request by IXI01**
Here is the chuffed code for the edit distance constraint propagator we published together with Peter in the AAAI 2020 paper "Explaining Propagators for String Edit Distance Constraints". It would be great if it could be included in future releases. If I should do anything else just let me know.

All the best,
Felix Winter

### CODE REVIEW

File: chuffed/globals/edit_distance.cpp [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+
+        // Step 2:
+        // Propagate lower bound to edit distance, and add explanation
+        //
+
+        // calc edit distance
+        int editDistanceLB = getEditDistanceLB();
+
+#ifndef NDEBUG
+//        updateDpMatrix(2 * seqSize * max_id_cost);
```
Comment (schutta) [inline chuffed/globals/edit_distance.cpp:164]: If this line and the following ones are not needed then please remove them from the code.

**Review by schutta — COMMENTED** [review]
Please check my comment in the file chuffed/globals/edit_distance.cpp.

**Comment by IXI01** [comment]
I removed now the commented code, thank you for pointing this out.

**Review by schutta — APPROVED** [review]

**Comment by Dekker1** [comment]
After this merge Chuffed no longer seems to compile in the MiniZinc pipeline. It fails with the following error (on all platforms):

```
[28/60] Building CXX object CMakeFiles/chuffed.dir/chuffed/globals/EdExplFinder.cpp.o
99FAILED: CMakeFiles/chuffed.dir/chuffed/globals/EdExplFinder.cpp.o 
100/Library/Developer/CommandLineTools/usr/bin/c++ -DHAS_PROFILER -I. -I/Users/tack/builds/ut-kUz_d/0/minizinc/vendor/chuffed -O3 -DNDEBUG -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -mmacosx-version-min=10.9 -std=gnu++11 -MD -MT CMakeFiles/chuffed.dir/chuffed/globals/EdExplFinder.cpp.o -MF CMakeFiles/chuffed.dir/chuffed/globals/EdExplFinder.cpp.o.d -o CMakeFiles/chuffed.dir/chuffed/globals/EdExplFinder.cpp.o -c /Users/tack/builds/ut-kUz_d/0/minizinc/vendor/chuffed/chuffed/globals/EdExplFinder.cpp
101In file included from /Users/tack/builds/ut-kUz_d/0/minizinc/vendor/chuffed/chuffed/globals/EdExplFinder.cpp:6:
102/Users/tack/builds/ut-kUz_d/0/minizinc/vendor/chuffed/chuffed/globals/EdExplFinder.h:10:10: fatal error: 'support/vec.h' file not found
103#include <support/vec.h>
104         ^~~~~~~~~~~~~~~
1051 error generated.
106[29/60] Building CXX object CMakeFiles/chuffed.dir/chuffed/globals/edit_distance.cpp.o
107FAILED: CMakeFiles/chuffed.dir/chuffed/globals/edit_distance.cpp.o 
108/Library/Developer/CommandLineTools/usr/bin/c++ -DHAS_PROFILER -I. -I/Users/tack/builds/ut-kUz_d/0/minizinc/vendor/chuffed -O3 -DNDEBUG -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -mmacosx-version-min=10.9 -std=gnu++11 -MD -MT CMakeFiles/chuffed.dir/chuffed/globals/edit_distance.cpp.o -MF CMakeFiles/chuffed.dir/chuffed/globals/edit_distance.cpp.o.d -o CMakeFiles/chuffed.dir/chuffed/globals/edit_distance.cpp.o -c /Users/tack/builds/ut-kUz_d/0/minizinc/vendor/chuffed/chuffed/globals/edit_distance.cpp
109In file included from /Users/tack/builds/ut-kUz_d/0/minizinc/vendor/chuffed/chuffed/globals/edit_distance.cpp:3:
110/Users/tack/builds/ut-kUz_d/0/minizinc/vendor/chuffed/chuffed/globals/EdExplFinder.h:10:10: fatal error: 'support/vec.h' file not found
111#include <support/vec.h>```

**Comment by IXI01** [comment]
I tried to compile chuffed from a fresh clone and ran into the same error. It seems that the following two includes in file EdExplFinder.h need to be replaced:

#include <support/vec.h>
#include <core/propagator.h>

should actually be

#include <chuffed/support/vec.h>
#include <chuffed/core/propagator.h>

Then it compiles for me. I am sorry for the inconvenience, for some reason it compiled in my build environment even with the old version.

**Pull request merged**